File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/request/RequestAcceptor.java
package net.mjduffin.risk.lib.usecase.request;

import net.mjduffin.risk.lib.usecase.GameState;
import net.mjduffin.risk.lib.usecase.PlayerOutput;

public interface RequestAcceptor {
    void receiveRequest(Request request);
    void registerPlayerOutput(PlayerOutput output);
    GameState getGameState();
}

File: risk-lib/src/main/kotlin/net/mjduffin/risk/lib/usecase/request/RequestAcceptor.kt
package net.mjduffin.risk.lib.usecase.request

import net.mjduffin.risk.lib.usecase.GameState

/**
 * Accepts a request.
 */
interface RequestAcceptor {
    fun receiveRequest(request: Request)
//    fun registerPlayerOutput(output: PlayerOutput)
    fun getGameState(): GameState
}